template/workflows/helm/.github/workflows/azure-kubernetes-service-helm.yml

```diff
@@ -108,34 +114,19 @@ jobs:
           admin: 'false'
           use-kubelogin: 'true'
 
-      # Runs Helm to create manifest files
-      - name: Bake deployment
-        uses: azure/k8s-bake@v2
-        with:
-          renderEngine: "helm"
-          helmChart: ${{ env.CHART_PATH }}
-          overrideFiles: ${{ env.CHART_OVERRIDE_PATH }}
-          overrides: ${{ env.CHART_OVERRIDES }}
-          helm-version: "latest"
-        id: bake
-
       # Checks if the AKS cluster is private
       - name: Is private cluster
         id: isPrivate
         run: |
           result=$(az aks show --resource-group ${{ env.CLUSTER_RESOURCE_GROUP }} --name ${{ env.CLUSTER_NAME }} --query "apiServerAccessProfile.enablePrivateCluster")
           echo "PRIVATE_CLUSTER=$result" >> "$GITHUB_OUTPUT"
 
-      # Deploys application based on manifest files from previous step
-      - name: Deploy application
-        uses: Azure/k8s-deploy@v4
-        with:
-          action: deploy
-          manifests: ${{ steps.bake.outputs.manifestsBundle }}
-          images: |
-            ${{ env.AZURE_CONTAINER_REGISTRY }}.azurecr.io/${{ env.CONTAINER_NAME }}:${{ github.sha }}
-          resource-group: ${{ env.CLUSTER_RESOURCE_GROUP }}
-          name: ${{ env.CLUSTER_NAME }}
-          private-cluster: ${{ steps.isPrivate.outputs.PRIVATE_CLUSTER }}
-          namespace: ${{ env.NAMESPACE }}
+      # Deploys application
+      - name: Deploy application on private cluster
+        if : steps.isPrivate.outputs.PRIVATE_CLUSTER == 'true'
+        run: az aks command invoke --resource-group ${{ env.CLUSTER_RESOURCE_GROUP }} --name ${{ env.CLUSTER_NAME }} --command "helm upgrade --wait -i -f ${{ env.CHART_OVERRIDE_PATH }} --set ${{ env.CHART_OVERRIDES }} automated-deployment ${{ env.CHART_PATH }}"
```

Comment:
have we tested this to ensure it works? do we not need to specify/mount the filepath using the --file argument on az aks command invoke?

Reply:
@jaiveerk I haven't tested the deployment yet, which filepath are you referring to?

Reply:
check this out - https://learn.microsoft.com/en-us/cli/azure/aks/command?view=azure-cli-latest#az-aks-command-invoke

from what I understand we need to specify files at the command-invoke level before they're mounted into the pod that actually executes the specified command for us.

I'm also skeptical about being able to run helm via az aks command invoke at all, so it may be worth doing a manual test (ie, generating the workflow and running it against a private cluster you create)

Reply:
why skepticism on being able to run Helm? This is a well documented feature

Reply:
ah gotcha - just found docs on it. concern re: needing to mount the files into the command still stand though

Reply:
I tested the workflow against both private and public cluster and it's working now. However, 'az aks command invoke' times out after around 5 min and it returns the exit code of az cli command not the helm upgrade command. So, the workflow is still showing as successful even though the helm upgrade failed.  
https://github.com/Azure/azure-cli/issues/22373
https://github.com/Azure/azure-cli/issues/23808